### PR TITLE
Feature/no results no entities alerts

### PR DESCRIPTION
**Pull request by simonychuang**
Part of the bigger UI redesign, this PR will improve the alert messages for different entities to distinguish between the cases where no such entity has been created yet vs no entities match a specific search criteria.

/jenkins-pr-deps https://github.com/Graylog2/graylog-plugin-enterprise/pull/4200
/nocl

## How Has This Been Tested?

Tested manually on local environment

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

/nocl

### CODE REVIEW

File: graylog2-web-interface/src/components/events/events/Events.jsx [hunk 1]
```diff
@@ -238,10 +238,20 @@ class Events extends React.Component {
...
 
+    const emptyListComponent = query ? (
+      <Alert bsStyle="info">
+        <Icon name="info-circle" />&nbsp;No {entity} found for the current search criteria.
+      </Alert>
+    ) : (
+      <Alert bsStyle="info">
+        <Icon name="info-circle" />&nbsp;No {entity} exist.
+      </Alert>
+    );
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/events/events/Events.jsx:253]: Let's use shared components for this, just to be able to adjust the layout for all similar cases in one place.
We already have the `EmptyResult` component, maybe we can rename it to something like `NoSearchResult` and create another component like `NoEntitiesExist`?

Comment (simonychuang) [reply graylog2-web-interface/src/components/events/events/Events.jsx:253]: sounds good to me. I'll work on getting that updated

Comment (simonychuang) [reply graylog2-web-interface/src/components/events/events/Events.jsx:253]: @linuspahl this should be complete now. I created a new component called `NoEntitiesExist` but I did notice there is the `EmptyEntity` component which has other usages that explain what the entity is and a link to create a new one. I chose not to unify the two but if you would like me to use `EmptyEntity` for all cases then I will probably need some texts for the different entities to explain what they are. Let me know what you would like to do.

File: graylog2-web-interface/src/components/common/NoEntitiesExist.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  className: PropTypes.string,
+};
+
+NoEntitiesExist.defaultProps = {
+  children: 'No entities exist.',
+  className: undefined,
+};
+
+export default NoEntitiesExist;
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/common/NoEntitiesExist.tsx:46]: Lets improve the way we display these component in out style guide by adding a `NoEntitiesExist.md` and `NoSearchResult.md` file and by adding a description for the `NoSearchRestult` component.

File: graylog2-web-interface/src/views/pages/DashboardsPage.tsx [hunk 3]
```diff
@@ -84,10 +98,14 @@ const DashboardsPage = () => {
 
         <Row className="content">
           <Col md={12}>
-            <DashboardList dashboards={list}
-                           pagination={pagination}
-                           handleSearch={handleSearch}
-                           handleDashboardDelete={handleDashboardDelete} />
+            {list !== undefined && list.length === 0 && !searchQuery
```
Comment (linuspahl) [inline graylog2-web-interface/src/views/pages/DashboardsPage.tsx:101]: It is a nitpick, but you could simplify this with `!!list?.length && !searchQuery`

File: graylog2-web-interface/src/components/streams/StreamList.jsx [hunk 4]
```diff
@@ -59,20 +57,12 @@ const StreamList = createReactClass({
   },
 
   render() {
-    if (this.props.streams.length > 0) {
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/streams/StreamList.jsx:62]: It looks like we are now always displaying the `EmptyEntity` component for the streams list.

**Review by linuspahl — CHANGES_REQUESTED** [review]

File: graylog2-web-interface/src/components/common/NoEntitiesExist.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ */
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+import { Alert } from 'components/bootstrap';
+import { Icon } from 'components/common';
+
+type Props = {
+  children: React.ReactNode,
+  className: string | null | undefined,
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/common/NoEntitiesExist.tsx:25]: Can't we just use `className?: string`?

File: graylog2-web-interface/src/components/common/NoEntitiesExist.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Component used to display a simple alert message for an empty entity in Graylog. For more complex messages that
+ * explain what that entity is and a link to create a new one please use <EmptyEntity>
+*/
+const NoEntitiesExist = ({ children, className }: Props) => (
+  <Alert className={`${className ?? ''} no-bm`}><Icon name="info-circle" />&nbsp;{children}</Alert>
+);
+
+NoEntitiesExist.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/common/NoEntitiesExist.tsx:37]: `PropTypes.node` includes `PropTypes.string`

File: graylog2-web-interface/src/components/common/NoSearchResult.tsx [hunk 7]
```diff
@@ -18,24 +18,29 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 
 import { Alert } from 'components/bootstrap';
+import { Icon } from 'components/common';
 
 type Props = {
   children: React.ReactNode,
   className: string | null | undefined,
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/common/NoSearchResult.tsx:25]: Not related to this PR, but let's use `className?: string` here as well. We used `string | null | undefined` when we migrated from flow to TS to ensure we use the exact same type definition.

File: graylog2-web-interface/src/components/events/events/Events.jsx (hunk 1 above)
Comment (linuspahl) [reply graylog2-web-interface/src/components/events/events/Events.jsx:253]: Sounds good!

File: graylog2-web-interface/src/components/outputs/OutputList.jsx [hunk 8]
```diff
@@ -33,6 +33,7 @@ class OutputList extends React.Component {
     types: PropTypes.object.isRequired,
   };
 
+  // eslint-disable-next-line class-methods-use-this
   _sortByTitle = (output1, output2) => {
     return naturalSort(output1.title.toLowerCase(), output2.title.toLowerCase());
   };
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/outputs/OutputList.jsx:39]: We can just move `sortByTitle` outside of the component and avoid using `eslint-disable`.

**Review by linuspahl — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**